src/app/visit/visit.component.ts

```diff
@@ -1,18 +1,45 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 
+import { Store } from '../models/store';
 import { Patient } from '../models/patient';
-
+import { StoreService } from '../services/store.service';
+import { SnackService, SNACK_TYPE } from '../services/snack.service';
 @Component({
     selector: 'app-visit',
     templateUrl: './visit.component.html',
     styleUrls: ['./visit.component.css'],
 })
-export class VisitComponent {
+export class VisitComponent implements OnInit {
+    constructor(
+        private storeService: StoreService,
+        private snacService: SnackService,
+    ) {}
     patient!: Patient;
+    store!: Store[];
     ngOnInit(): void {
         this.patient = history.state;
+        this.storeService.getStore().subscribe({
+            next: (data) => (
+                (this.store = data),
+                (this.store = data.sort((a, b) => a.id - b.id))
```

Comment:
`(this.store = data),` remove this part, you are assigning it the value with sort to store in a line below